trieste/acquisition/combination.py

```diff
@@ -130,19 +127,42 @@ def _reduce(self, inputs: Sequence[TensorType]) -> TensorType:
         return tf.add_n(inputs)
 
 
-class Product(Reducer):
+class Product(Reducer[ProbabilisticModelType]):
     """
     :class:`Reducer` whose resulting acquisition function returns the element-wise product of the
     outputs of constituent acquisition functions.
     """
 
-    def __repr__(self) -> str:
-        """"""
-        return f"Product({self._repr_builders()})"
-
     def _reduce(self, inputs: Sequence[TensorType]) -> TensorType:
         """
         :param inputs: The outputs of each acquisition function.
         :return: The element-wise product of the ``inputs``.
         """
         return tf.reduce_prod(inputs, axis=0)
+
+
+class Map(Reducer[ProbabilisticModelType]):
+    """
+    :class:`Reducer` that accepts just one acquisition function builder and applies a
```

Comment:
can you add more details here, some example perhaps - a user would not easily see what is this for based on this description...